### PR TITLE
WIP: ILR transform on tensors

### PR DESCRIPTION
**Pull request by mortonjt**
This addresses https://github.com/qiime2/q2-gneiss/issues/74

The idea here we want to be able to apply the ILR transform directly to the outputs of Bayesian differential abundance tools such as Birdman.  Doing so will allow for robust hypothesis testing, uncertainty estimation of balances in addition to providing an additional link between differential abundance tools and Empress.

Of course, this all depends on finalizing the upcoming tensor types https://github.com/gibsramen/BIRDMAn/issues/26
In the most ideal scenario, this would be included in the August qiime2 release, since we are aiming to include this functionality in upcoming manuscripts / plugins.

### CODE REVIEW

**Comment by thermokarst** [comment]
Hey @mortonjt - the PR submission cutoff for the 2021.4 release was last week - I don't think we'll be able to squeeze this into this release. The next release is slated for August.

**Comment by mortonjt** [comment]
@thermokarst the August release timeline sounds perfect :)

File: q2_gneiss/composition/_method.py [hunk 1]
```diff
@@ -48,6 +50,51 @@ def ilr_phylogenetic_differential(
...
 
+def ilr_phylogenetic_posterior_differential(
+        posterior: xr.Dataset, tree: skbio.TreeNode,
+        minimax_filter: bool = True
+) -> (xr.DataArray, pd.DataFrame, skbio.TreeNode):
+    dataset, trimmed_tree = _xarray_match_tips(posterior, tree, 'features')
+    trimmed_tree = rename_clades(trimmed_tree)
+    # TODO: watch out for this indexing with diff. The ordering of the dims
+    # could mess things up here.
+    posterior_data = np.array(dataset['diff'].values)
```
Comment (mortonjt) [inline q2_gneiss/composition/_method.py:61]: we should replace this with a pandas pivot so that the dimension order doesn't matter.

**Comment by mortonjt** [comment]
Just raised a PR here on the tensor types: https://github.com/qiime2/q2-types/pull/261
The pull request here depends on getting those tensor types in.

**Comment by mortonjt** [comment]
@thermokarst just a heads up on this.  After talking to @ebolyen and @nbokulich, I think the most reasonable approach is to address https://github.com/qiime2/q2-types/pull/261 by first creating a separate [q2-bayes-types](https://github.com/mortonjt/q2-bayes-types) repository with the appropriate tensor types in place.  That repository still needs to be fleshed out, so I don't think that can be done before the next release.

**Comment by lizgehret** [comment]
Hey @mortonjt - we are doing some PR triage and review right now, and I looked over this PR along with the one from q2-types you linked above ([#261](https://github.com/qiime2/q2-types/pull/261)). Are you still building that custom repo for the types you discussed with @nbokulich? And if so, are the changes in this PR still current?

**Comment by mortonjt** [comment]
Hi @lizgehret thank you for following up on this.  I had some q2type work on https://github.com/mortonjt/q2-types/commit/060acfe805c233d4d2946eaa0fca15e81186db65

But the effort was proving to be too much for me to work on at this point -- I would like to revisit it at a later date; but it will require me to either reprioritize projects or have someone else lead.

**Comment by ebolyen** [comment]
No worries @mortonjt! I'm going to close this, but feel free to re-open if the situation changes.

**Pull request closed without merging**